src/sdk/Validator.ts

```diff
@@ -26,83 +24,72 @@ export const validate = (application: Application): string[] => {
       });
     }
     if (!utils.urlValidation(application.api.token)) {
-      issues.push(
-        'Manifest Api section needs to have a valid token endpoint url. Value: ' +
-          application.api.token
-      );
+      issues.push('Manifest Api section needs to have a valid token endpoint url. Value: ' + application.api.token);
     }
 
     if (!application.api.applicationId) {
       issues.push('Manifest Api section needs to have applicationId value.');
     }
 
+    if (!application.api.client) {
+      issues.push('Manifest Api section needs to have client value.');
+    } else if (!application.api.client.id) {
+      issues.push('Manifest Api section needs to have client.id value.');
+    }
+
     if (!utils.urlValidation(application.api.redirectUri)) {
```

Comment:
This should validate with regard to `application.api.redirectUris`. 
E.g. now it will always fail on missing `redirectUri` even though the `redirectUris` is filled.

Reply:
Added a condition